orchagent/fgnhgorch.cpp

```diff
@@ -718,9 +818,16 @@ bool FgNhgOrch::setActiveBankHashBucketChanges(FGNextHopGroupEntry *syncd_fg_rou
                 }
                 else
                 {
-                    uint32_t last_elem = map_entry->at((*map_entry).size() - 1);
+                    if ((*map_entry).size() == exp_size_before_bkt_rem)
+                    {
+                        SWSS_LOG_INFO("%s Buckets size already at %d, don't remove more",
+                                it->to_string().c_str(), exp_size_before_bkt_rem);
+                        it = active_nh_list.erase(it);
```

Comment:
There are 2 bugs introduced by this new change:
1. It will result in double erasal from active_nh_list(here as well as deletion per the original logic below), thereby causing you to delete 2 entries in 1 loop, and removing an entry you should not be deleting from the active_nh_list.
2. You are still allocating a bucket from this entry despite the delete performed here, so don't think this logic is doing what it is intended to do.

Reply:
I've rewritten this logic and this code fragment is no longer needed.